### PR TITLE
Signature Fix for UTF-8 Encoding and Field Overlap

### PR DESCRIPTION
**Pull request by SchulzeS**
The sorting of keys by UTF8 Encoding leads to an unstable sorting, if the key contains invalid codepoints. This is now solved by using the plain byte array and the usage of BASE64 Encoding for the sorting. The change contains also new seperators in the singature calculation to avoid overlaps between different fields.

### CODE REVIEW

**Comment by alstiefel** [comment]
Please do not close this PR until we get political approval. This is a breaking change.

**Review by dfischer-tech — APPROVED** [review]

**Review by mschulte-tsi — APPROVED** [review]

**Pull request merged**
